File: py/arrays and searching sorting/employee_salary_selection-bubble.py
# List of employee salaries
salaries = [55000.5, 72000.75, 48000.25, 91000.0, 66000.4,
            83000.9, 39000.0, 102000.25, 58000.1, 47000.8]

n = len(salaries)

# ---- SELECTION SORT ----
selection_sorted = salaries.copy()
for i in range(n - 1):
    min_index = i
    for j in range(i + 1, n):
        if selection_sorted[j] < selection_sorted[min_index]:
            min_index = j
    # Swap
    selection_sorted[i], selection_sorted[min_index] = selection_sorted[min_index], selection_sorted[i]

print("Salaries sorted using Selection Sort (Ascending):")
print(selection_sorted)
print()

# ---- BUBBLE SORT ----
bubble_sorted = salaries.copy()
for i in range(n - 1):
    for j in range(0, n - i - 1):
        if bubble_sorted[j] > bubble_sorted[j + 1]:
            # Swap
            bubble_sorted[j], bubble_sorted[j + 1] = bubble_sorted[j + 1], bubble_sorted[j]

print("Salaries sorted using Bubble Sort (Ascending):")
print(bubble_sorted)
print()

# ---- TOP 5 HIGHEST SALARIES ----
# The list is already sorted in ascending order, so take last 5 elements
top_5 = selection_sorted[-5:]

print("Top 5 Highest Salaries:")
for salary in reversed(top_5):  # Print from highest to lowest
    print(salary)

File: ds practical/cpp/arrays and searching sorting algo/employee_salary_selection-bubble.cpp
#include <iostream>
using namespace std;

int main() {
    double salaries[10] = {55000.5, 72000.75, 48000.25, 91000.0, 66000.4,
                           83000.9, 39000.0, 102000.25, 58000.1, 47000.8};

    int n = 10;
    double temp;

    // ---- SELECTION SORT ----
    for (int i = 0; i < n - 1; i++) {
        int minIndex = i;
        for (int j = i + 1; j < n; j++) {
            if (salaries[j] < salaries[minIndex]) {
                minIndex = j;
            }
        }
        // Swap
        temp = salaries[i];
        salaries[i] = salaries[minIndex];
        salaries[minIndex] = temp;
    }

    cout << "Salaries sorted using Selection Sort (Ascending):\n";
    for (int i = 0; i < n; i++) {
        cout << salaries[i] << "  ";
    }
    cout << endl << endl;

    // ---- Reset array for Bubble Sort ----
    double salaries2[10] = {55000.5, 72000.75, 48000.25, 91000.0, 66000.4,
                            83000.9, 39000.0, 102000.25, 58000.1, 47000.8};

    // ---- BUBBLE SORT ----
    for (int i = 0; i < n - 1; i++) {
        for (int j = 0; j < n - i - 1; j++) {
            if (salaries2[j] > salaries2[j + 1]) {
                temp = salaries2[j];
                salaries2[j] = salaries2[j + 1];
                salaries2[j + 1] = temp;
            }
        }
    }

    cout << "Salaries sorted using Bubble Sort (Ascending):\n";
    for (int i = 0; i < n; i++) {
        cout << salaries2[i] << "  ";
    }
    cout << endl << endl;

    // ---- TOP 5 HIGHEST SALARIES ----
    cout << "Top 5 Highest Salaries:\n";
    for (int i = n - 1; i >= n - 5; i--) {
        cout << salaries[i] << endl;
    }

    return 0;
}
